src/profile/dto/update-profile.dto.ts

```diff
@@ -1,4 +1,7 @@
-import { PartialType } from '@nestjs/swagger';
-import { CreateProfileDto } from './create-profile.dto';
-
-export class UpdateProfileDto extends PartialType(CreateProfileDto) {}
+import { ApiPropertyOptional } from '@nestjs/swagger';
+import { IsEmail } from 'class-validator';
+export class UpdateProfileDto {
+  @IsEmail()
+  @ApiPropertyOptional()
+  email?: string;
```

Comment:
You could use the IsOptional decorator from the class-validator package instead of the ? after email.

Secondly, I dont think it should be optional since it is the only property.